src/template/transaction/setvarcommands.ts

```diff
@@ -23,7 +23,7 @@ export async function getWalletAddress(
   _ctx: TxTemplateContext,
   options: z.infer<typeof SetVarGetWalletAddressOpts>
 ): Promise<string> {
-  if (options.index) {
+  if (options.index != null) {
```

Comment:
This is the template small fix. The index might be 0, which wasn't entering the `if`